components/remote-environment-broker/internal/broker/middleware.go

```diff
@@ -4,16 +4,48 @@ import (
 	"net/http"
 
 	osb "github.com/pmorie/go-open-service-broker-client/v2"
+	"github.com/sirupsen/logrus"
 )
 
 // OSBContextMiddleware implements Handler interface
-type OSBContextMiddleware struct{}
+type OSBContextMiddleware struct {
+	brokerFlavor brokerFlavorProvider
+	log          *logrus.Entry
```

Comment:
please use interface `logrus.FieldLogger` instead of pointer to struct, same for the arg in `NewOsbContextMiddleware` constructor